### PR TITLE
feat: add lambda client

### PR DESCRIPTION
**Pull request by vmaerten**
We duplicate lambda in each project

### CODE REVIEW

File: src/lambda.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            .client
+            .invoke()
+            .function_name(function_name)
+            .payload(Blob::new(payload))
+            .log_type(Tail)
+            .send()
+            .await
+            .unwrap();
+        LambdaResult {
+            base64_logs: res.log_result.unwrap_or_else(|| "".to_string()),
```
Comment (tvincent2) [inline src/lambda.rs:51]: I think you don't really need `unwrap_or_else` and a closure here, `unwrap_or` should be enough:
```suggestion
            base64_logs: res.log_result.unwrap_or("".to_string()),
```

File: Cargo.toml [hunk 2]
```diff
@@ -15,10 +15,12 @@ serde_dynamo = { version = "4.0.2", features = [ "aws-sdk-dynamodb+0_13" ] }
 aws-config = { version =  "0.13.0", default-features = false, features = [ "native-tls" ] }
 aws-types = { version =  "0.13.0" }
 aws-sdk-dynamodb = { version = "0.13.0", default-features = false, features = [ "native-tls" ] }
-
-
+aws-sdk-lambda = { version = "0.13.0", default-features = false, features = ["native-tls"]}
+aws-smithy-types = "0.34.1"
```
Comment (tvincent2) [inline Cargo.toml:19]: There have been several newer versions, is there a reason to keep this one?

```suggestion
aws-smithy-types = "0.43.0"
```

File: src/lambda.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+pub struct LambdaResult {
+    base64_logs: String,
+    payload: Vec<u8>,
+}
+
+impl LambdaResult {
+    pub fn get_logs(&self) -> String {
+        std::str::from_utf8(&base64::decode(&self.base64_logs).unwrap())
+            .unwrap()
+            .to_string()
```
Comment (tvincent2) [inline src/lambda.rs:24]: You could directly build a `String` here:
```suggestion
        String::from_utf8(base64::decode(&self.base64_logs).unwrap())
            .unwrap()
```

Comment (vmaerten) [reply src/lambda.rs:24]: Thanks!

File: Cargo.toml (hunk 2 above)
Comment (vmaerten) [reply Cargo.toml:19]: I don't think so !

**Review by tvincent2 — APPROVED** [review]

File: src/lambda.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    payload: Vec<u8>,
+}
+
+impl LambdaResult {
+    pub fn get_logs(&self) -> String {
+        String::from_utf8(base64::decode(&self.base64_logs).unwrap())
+            .unwrap()
+    }
+    pub fn get_payload(&self) -> String {
+        std::str::from_utf8(&self.payload).unwrap().to_string()
```
Comment (tvincent2) [inline src/lambda.rs:26]: Sorry I didn't notice that you could make a similar change than in `get_logs` here:
```suggestion
        String::from_utf8(&self.payload.clone()).unwrap()
```
Note that the `clone()` was hidden in the call to `to_string()` anyway.

Comment (vmaerten) [reply src/lambda.rs:26]: thanks for all suggestion

**Pull request merged**